monix-eval/shared/src/test/scala/monix/eval/TaskMemoizeOnSuccessSuite.scala

```diff
@@ -645,13 +646,13 @@ object TaskMemoizeOnSuccessSuite extends BaseTestSuite {
     assertEquals(task, task.memoizeOnSuccess)
   }
 
-  testAsync("local.write.memoize works") { _ =>
+  testAsync("local.write.memoizeOnSuccess works") { _ =>
     import monix.execution.Scheduler.Implicits.global
     implicit val opts = Task.defaultOptions.enableLocalContextPropagation
 
     val task = for {
       local <- TaskLocal(0)
-      v1    <- (local.write(100).flatMap(_ => local.read)).memoizeOnSuccess
+      v1    <- local.write(100).flatMap(_ => local.read).memoizeOnSuccess
```

Comment:
Just for the sake of inner peace, can you execute the memoized task two or more times to ensure that it doesn't change the local state after first time?

Reply:
I upgraded to
```scala
val task = for {
    local <- TaskLocal(0)
    memoizeTask = local.write(100).flatMap(_ => local.read).memoize
    v1    <- memoizeTask
    _     <- Task.shift
    v2    <- local.read
    _     <- local.write(200)
    _     <- memoizeTask
    _     <- Task.shift
    v3    <- local.read
} yield (v1, v2, v3)

for (v <- task.runToFutureOpt) yield {
    assertEquals(v, (100, 100, 200))
}
```

and seems to still work. :D 

Initially I changed this test to say that `memoize` + `local` don't work, updated scaladoc etc. and then to my surprise the test was failing because it actually still worked... I guess `restoreLocals` can still be useful if we want to customize locals for `Task`